app/controllers/api/v1/merchants_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Api::V1::MerchantsController < Api::V1::ApplicationController
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+  before_action :restrict_access
+
+  def index
+    merchants = Store.all
+    render json: merchants
+  end
+
+  private
+
+    def restrict_access
+      authenticate_or_request_with_http_token do |token, options|
+        ENV['API_KEY'] == token
```

Comment:
Looks good. If we need to make this externally consumable we can change it back.